### ISSUE TITLE
Implement basic performance benchmark

### ISSUE THREAD

**Issue by drewoldag** [lincc-frameworks/fibad#965]
We received a report that the time needed to run training with Hyrax was ~2-3x longer compared to training with pure PyTorch.

It would be good to benchmark our performance. The most obvious case to me is training the HyraxCNN using the HyraxCifar dataset.

PyTorch documentation provides a very straightforward implementation (in fact that is what HyraxCNN and HyraxCifar as based on) here: https://docs.pytorch.org/tutorials/beginner/blitz/cifar10_tutorial.html

A highlevel understanding of the difference in training time for Hyrax vs. PyTorch would be very helpful. Ultimately we would want to be able to get very fine grained so that we can improve Hyrax performance as needed, but anything is better than nothing. As a first pass we would want to know whether there is a significant disparity (on the order of 2-3x slower) or not.

**Comment by Graciaaa3** [comment]
Work so far & things to note:
- Implemented `benchmarks_self_hosted/run_timebench.py` to benchmark the Hyrax and PyTorch pipeline using a CNN model + Cifar10 dataset. 
- Configurable training parameters include: 
   - train_fraction, 
   - epochs, 
   - batch_size, 
   - num_workers - increasing this does not appear to improve performance.
   - lr, 
   - repeats - used to run multiple trials and report the mean and std.
- Each benchmark run saves a json result containing runtime, accuracy, slowdown factors, environment information, git commit, etc. Files are currently formatted as `<short_commit>_<YYYMMDD_HHMMSS>.json`. This allows multiple unique results to be stored for the same day.
- The github action `.github/workflows/pipeline_time_bench.yml` is responsible for running the benchmark and publishing the results to `benchmarks` branch under a directory other than `_results` (asv manages this). Currently, the name of the new dir is `gondor_results/`. The following paths in the workflow are currently Gondor-specific and should be updated if the benchmark is run in a different environment:
```
# path to downloaded cifar dataset
# currently gondor-specific path, should be changed for other environments
export CIFAR_DIR="/astro/store/shiren/wqy37/datasets"

# path for saving the result
# currently gondor-specific path, should be changed for other environments
export RESULT_PATH="/astro/store/shiren/wqy37/results"
```
ASV tracking
- Added `track_*` functions in `benchmarks/track_pipeline_benchmarks.py` to publish the custom benchmark data to ASV for plotting. The `track_*` function simply return a value, they do not perform and actual benchmarking runs.
- The tracking functions read the slowdown metrics from the json results generated by the pipeline benchmark. Currently I have: 
  - `track_train_slowdown`
  - `track_infer_slowdown`
  - `track_total_slowdown`
  - `track_test` - initial testing function outside of the class.
- The first three tracking functions are wrapped in a class that loads a json result once through `setup_cache()`. The following path is currently hard-coded for testing and should be updated to dynamically select the appropriate result file:
```
# TODO: "gondor_results" is the directory to store results from self-hosted runner on gondor
base_dir = Path(__file__).resolve().parent / "gondor_results"
# TODO: currently hard coded to test file loading only
self.json_path = base_dir / "2026-07-24T22:08:39.409085+00:00.json"
```
